src/ChunkMap.cpp

```diff
@@ -1313,21 +1313,19 @@ void cChunkMap::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_
 				a_Entity.TakeDamage(dtExplosion, nullptr, static_cast<int>((1 / std::max(1.0, DistanceFromExplosion.Length())) * 8 * ExplosionSizeInt), 0);
 			}
 
-			double Length = DistanceFromExplosion.Length();
-			if (Length <= ExplosionSizeInt)  // Entity is impacted by explosion
-			{
-				float EntityExposure = a_Entity.GetExplosionExposureRate(ExplosionPos, static_cast<float>(a_ExplosionSize));
+			float EntityExposure = a_Entity.GetExplosionExposureRate(ExplosionPos, static_cast<float>(a_ExplosionSize));
 
-				// Exposure reduced by armor
-				EntityExposure = EntityExposure * (1.0f - a_Entity.GetEnchantmentBlastKnockbackReduction());
+			// Exposure reduced by armor
+			EntityExposure = EntityExposure * (1.0f - a_Entity.GetEnchantmentBlastKnockbackReduction());
 
-				double Impact = (1 - ((Length / ExplosionSizeInt) / 2)) * EntityExposure;
+			auto Impact = std::pow(std::max(0.2, DistanceFromExplosion.Length()), -1);
+			Impact *= EntityExposure * ExplosionSizeInt / 5.0;
 
-				DistanceFromExplosion.Normalize();
-				DistanceFromExplosion *= Impact;
+			DistanceFromExplosion.Normalize();
+			DistanceFromExplosion *= Vector3d{Impact, 0.0, Impact};
+			DistanceFromExplosion.y += 0.3 * Impact;
 
-				a_Entity.SetSpeed(DistanceFromExplosion);
-			}
+			a_Entity.SetSpeed(DistanceFromExplosion);
```

Comment:
This just seems wrong - how does setting the speed for all entities even those outside of the explosion radius _reduce_ the interference? This looks like it should set speed for all entities outside of the explosion radius to zero, which is not really a desired outcome.

Reply:
Player speeds aren't really updated properly so sending 0 is preferable to sending junk.

Reply:
I could make this player only I suppose.